lib/ts/recipe/session/recipeImplementation.ts

```diff
@@ -139,104 +173,166 @@ export default function getRecipeInterface(
+                if (response.accessToken !== undefined) {
+                    setFrontTokenInHeaders(
+                        res,
+                        response.session.userId,
+                        response.accessToken.expiry,
+                        response.session.userDataInJWT
+                    );
+                    setToken(
+                        config,
+                        res,
+                        "access",
+                        response.accessToken.token,
+                        // We set the expiration to 100 years, because we can't really access the expiration of the refresh token everywhere we are setting it.
+                        // This should be safe to do, since this is only the validity of the cookie (set here or on the frontend) but we check the expiration of the JWT anyway.
+                        // Even if the token is expired the presence of the token indicates that the user could have a valid refresh
+                        // Setting them to infinity would require special case handling on the frontend and just adding 10 years seems enough.
+                        Date.now() + 3153600000000,
+                        requestTransferMethod
+                    );
+                    accessTokenString = response.accessToken.token;
+                }
+                logDebugMessage("getSession: Success!");
+                const session = new Session(
+                    helpers,
+                    accessTokenString,
+                    response.session.handle,
                     response.session.userId,
-                    response.accessToken.expiry,
-                    response.session.userDataInJWT
+                    response.session.userDataInJWT,
+                    res,
+                    req,
+                    requestTransferMethod
                 );
-                attachAccessTokenToCookie(config, res, response.accessToken.token, response.accessToken.expiry);
-                accessToken = response.accessToken.token;
-            }
-            logDebugMessage("getSession: Success!");
-            const session = new Session(
-                helpers,
-                accessToken,
-                response.session.handle,
-                response.session.userId,
-                response.session.userDataInJWT,
-                res
-            );
 
-            return session;
+                return session;
+            } catch (err) {
+                // We can clear the session from all transfer methods here, since we received a valid, non-expired token in the current method
+                if (err.type === STError.UNAUTHORISED) {
+                    logDebugMessage(
+                        `getSession: Clearing ${requestTransferMethod} because of UNAUTHORISED response from getSession`
+                    );
+                    clearSession(config, res, requestTransferMethod);
+                }
+                throw err;
+            }
```

Comment:
- why do we need to explicitly do this here? Won't the errorHandler just catch the `UNAUTHORISED` and clear the session as well?
- The comment says "We can clear the session from all transfer methods here". But when you call `clearSession`, you pass in the specific `requestTransferMethod`.
- Should't you check for if clearTokens is true in the error?

Reply:
I haven't updated these during the merge (that introduced the errors clearing everything). I'll just remove these sections.